src/CrossScopeValidator.ts

```diff
@@ -236,4 +239,40 @@ export class CrossScopeValidator {
             }
         }
     }
+
+    /**
+     * Add a diagnostic (to the first scope) that will have `relatedInformation` for each affected scope
+     */
+    private addMultiScopeDiagnostic(scope: Scope, diagnostic: BsDiagnosticWithOrigin) {
+        diagnostic = this.multiScopeCache.getOrAdd(
+            `${diagnostic.file?.srcPath}-${diagnostic.code}-${diagnostic.message}-${util.rangeToString(diagnostic.range)}`,
+            () => {
+                if (!diagnostic.relatedInformation) {
+                    diagnostic.relatedInformation = [];
+                }
+
+                scope.addDiagnostic(diagnostic);
+                return diagnostic;
+            }
+        );
+        if (isXmlScope(scope) && scope.xmlFile?.srcPath) {
+            diagnostic.relatedInformation.push({
+                message: `In component scope '${scope?.xmlFile?.componentName?.text}'`,
+                location: util.createLocation(
+                    URI.file(scope.xmlFile.srcPath).toString(),
+                    scope?.xmlFile?.ast?.componentElement?.getAttribute('name')?.tokens?.value?.range ?? util.createRange(0, 0, 0, 10)
+                )
+            });
+        } else {
+            diagnostic.relatedInformation.push({
+                message: `In scope '${scope.name}'`,
+                location: util.createLocation(
+                    URI.file(diagnostic.file.srcPath).toString(),
+                    diagnostic.range
+                )
+            });
+        }
+    }
+
+    private multiScopeCache = new Cache<string, BsDiagnosticWithOrigin>();
```

Comment:
We will need to clear this cache between each scope validation (similar to what we do in ScopeValidator)